Sources/WhoopDIKit/Container/ThreadSafeDependencyGraph.swift

```diff
@@ -0,0 +1,35 @@
+import Foundation
+
+final class ThreadSafeDependencyGraph: Sendable {
+    private let lock = NSRecursiveLock()
+    nonisolated(unsafe) private let serviceDict: ServiceDictionary<DependencyDefinition> = .init()
```

Comment:
let's not make this `nonisolated(unsafe)`, but instead make the class `@unchecked Sendable` since that is more accurate to our usage here

Reply:
Could you elaborate on why `unchecked Sendable` is preferable here? My rationale is that I wanted to scope this check down to just this variable since that is the only thing which is unsafe. If we were to make that a Sendable in the future we could remove this all together.

Reply:
In the way I see it, we are providing the guarantee that the `ThreadSafeDependencyGraph` is sendable, since we are providing locking around the non-sendable type. This to me means that this type is `@unchecked Sendable` because we are providing this guarantee to tell the compiler. From my understanding, `nonisolated(unsafe)` is more of an escape hatch where we _know_ something won't race, but the compiler can't reason about it (but we don't really need to do anything, where here we are doing something to make it Sendable)